src/extensions/default/BrambleUrlCodeHints/main.js

```diff
@@ -144,29 +144,58 @@ define(function (require, exports, module) {
             self.cachedHints.unfiltered = [];
             self.cachedHints.imageUrl = self.imageUrl;
 
-            directory.getContents(function (err, contents) {
-                var currentDeferred, entryStr, syncResults;
-                if(err) {
-                    return;
-                }
+            function deepWalk(path, parentPath, callback) {
+                var directory = FileSystem.getDirectoryForPath(Path.join(parentPath, path));
 
-                contents.forEach(function (entry) {
-                    if (!ProjectManager.shouldShow(entry)) {
-                        return;
+                directory.getContents(function (err, contents) {
+                    if(err) {
+                        return callback(err);
                     }
 
-                    // convert to doc relative path
-                    entryStr = queryDir + entry._name;
-                    // TODO: should do a recursive walk of the files
-                    if (entry._isDirectory) {
-                        return;
-                    }
+                    Async.doSequentially(contents, function(entry) {
+                        var deferred = new $.Deferred();
+                        var filename;
+
+                        if (!entry._isDirectory) {
+                            if (ProjectManager.shouldShow(entry) &&
+                                Content.isImage(Path.extname(entry._name))) {
```

Comment:
Interestingly, I see these image paths in the path hint dropdown even if i'm putting in a `src` for a `script` tag. So we might want to do file type checking here?